### DISCUSSION TITLE
Help diagnosing PCM5102 DAC that is making no sound

### DISCUSSION THREAD

**Discussion by viragored** [probonopd/MiniDexed#608] [Q&A] [answered]
Hi - I'm not very experienced at building and soldering, but I have got a working MiniDexed. Everything works as it should on a Pi 4 with sound via external HDMI adapter. I have now added a PCM5102 DAC, but am getting no sound from it. I'd appreciate any pointers to what I can try next to get sound. Steps so far:

- Printed a case, installed LCD screen with backpack and logic level converter, attached capacitors to rotary encoder, connected to Pi 4 GPIO pins, plugged in HDMI sound adapter, edited minidexed.ini, and all works as it should
- Soldered wires to DAC as shown in [Hardware](https://github.com/probonopd/MiniDexed/wiki/Hardware) (Didn't discover the SCK jumper on top until after I'd soldered a wire to GND!)
- Edited minidexed.ini, Connected to Pi 4 - no sound. Amplifier connection seems to be working because holding the jack socket contacts makes a hum
- Googled, and discovered jumpers on the DAC have to be soldered. Picture included showing my soldered jumpers
- Connected to Pi 4 - still no sound. Used multimeter to check the solder on the jumpers is not bridging to the unused solder pad.
- Confirmed 5V present at VIN + GND. No power status LED lights up on the DAC. I am sharing 5V GPIO pin 4 with the logic level converter; the LCD screen is powered from GPIO pin 2

[My DAC](https://epartners.co.nz/products/vx3012) is from a local-to-me supplier, and has the Jumper pad labels in a weird fashion, so I hope I've got the actual jumpers correct. 
![DAC jumpers soldered](https://github.com/probonopd/MiniDexed/assets/28804416/b0f582d0-e3c8-4723-928a-0d98e2b0d97d)

I'd be very grateful for any pointers to things I can try to work out if the DAC is faulty, my soldering or wiring is faulty, and hopefully get it making sound.

**Comment by Banana71** [comment]
Hello viragord,

Everything looks good to me so far. I've never noticed that anyone here had a defective DAC.
However, no power on status LED is strange and the first thing I would do is check the soldering points of the power supply on the DAC.
If that doesn't help, can you please take more pictures? The front of the DAC, the connections of the Raspberry PI.

**Comment by RK11111111111** [comment]
You said you edited the minidexed.ini.  I presume that includes enabling i2s.  Can you check to see if sound is coming from the on-board jack?

I am interested in these other audio adapters, I have seen them and it's hard to believe these dacs dominate the cost of these synth projects.  I've been using raspiaudio DACs, but they are a hat, so it's not as easy to work with the pins on the pi.

**Comment by diyelectromusic** [comment]
When you say "I am sharing 5V GPIO pin 4 with the logic level converter" what is the converter used for?  Those GY-PCM5102 modules don't need one for 5V power but 3V3 operation in my experience - is that for the LCD?

Aside: the jumpers and pin labels match the ones I've always used, so that should be ok.

How are you driving your MiniDexed - over MIDI?  Is it worth enabling MIDI debug to check messages are getting through? It might be worth posting a photo of the boot log (on a HDMI monitor) and we might need a photo/diagram of your connections...

Kevin

**Comment by viragored** [comment]
Thanks for all the suggestions. I've taken some photos which I hope will help. Sorry they're not great, it's a bit dark now. I can repeat in daylight if that would help.
Things I've also done:

1. Re-soldered all the wire connections to the DAC
2. Pulled all the DuPont connectors and made sure they are a tight fit on their pins
3. Changed minidexed.ini to use each of the sound settings (by adding or removing the # on the SoundDevice lines, using Mousepad text editor on a R Pi)

Results of the enabled minidexed.ini selections (ie no # for the SoundDevice) are:

- i2s only = no sound
- i2s + pwm = no sound
- i2s + pwm + hdmi = no sound
- pwm only = onboard jack plays in headphones
- pwm + hdmi = onboard jack plays in headphones, no sound from HDMI adapter
- hdmi alone = sound through amp from HDMI adapter

I'm using an Alesis 49-keyboard midi controller plugged into the R Pi USB.
Hopefully the picture will show what I mean by sharing the +5V and GND - I've spliced the wires together, run solder into the joint and covered it with heatshrink. And when all the bits are in the 3D-printed case, I'm using some electrical insulation tape to cover the components to ensure there are no unwanted electrical contacts.
The logic level converter is to protect the Pi GPIO from 5V coming back from the LCD backpack; the LCD gets its power from GP Pin 2.
The boot logs show what happened when booting with i2s only, i2s+pwm+hdmi, and hdmi only. (The file names explain which is which, but they seem to disappear during upload).
And there's still no power status LED shining on the DAC, even though my meter shows 5V present across the VIN and GND connections.

![DAC top](https://github.com/probonopd/MiniDexed/assets/28804416/e02f56f0-a240-4e05-b72e-498f00ed28f6)
![Backpack Rotary encoder](https://github.com/probonopd/MiniDexed/assets/28804416/ed3058ad-6857-4a60-b7c7-5574ff4b29b7)
![Logic leval converter](https://github.com/probonopd/MiniDexed/assets/28804416/27642f4f-77f0-4229-9314-5be7ada7037f)
![GPIO 2 - 40](https://github.com/probonopd/MiniDexed/assets/28804416/96d8b262-6784-4911-8b2d-c7141a5aa297)
![GPIO 1 - 39](https://github.com/probonopd/MiniDexed/assets/28804416/249ae15e-e54f-41be-88ab-f1a7dc300e15)
![Boot i2s](https://github.com/probonopd/MiniDexed/assets/28804416/6dda0258-cef1-4c5d-b314-9b48e48adc53)
![Boot i2s pwm hdmi](https://github.com/probonopd/MiniDexed/assets/28804416/fd5ccbbf-a6a9-4189-803d-6b007af35dae)
![Boot HDMI working](https://github.com/probonopd/MiniDexed/assets/28804416/3005cd51-fcd6-4c85-bd66-40f613869bef)

**Comment by Banana71** [comment]
The miniDexed can only control one sound device at the same time. The minidexed.ini could also look like this:
```
#
# minidexed.ini
#

# Sound device ( i2s ; pwm ; hdmi )
SoundDevice=i2s
SampleRate=48000
#ChunkSize=256
DACI2CAddress=0
ChannelsSwapped=0
# Engine Type ( 1=Modern ; 2=Mark I ; 3=OPL )
EngineType=1
```

I have an identical DAC only connected with 2 wires, 5V (VIN) and GND (GND), and the power LED lights up. Maybe this way you can rule out whether the DAC is defective.

Peter

**Comment by viragored** [comment]
Good suggestion! It's a bit of a lash-up, but here is a pic of the DAC powered directly from USB 5V. All DuPont connectors are disconnected from the Pi. The logic level converter is also being powered from the same source, and the two soldered signal wires remain between the converter and the LCD backpack. No power is going to the LCD display. The power status LED on the DAC remains unlit :-(

Does that mean the DAC is dead? If so, I think it's worth my ordering a bunch from Ali Express rather than the regular supplier (who seems to be currently the only one in the country with any of these boards in stock). I can live with the working HDMI sound adapter for a while.

![Power no light](https://github.com/probonopd/MiniDexed/assets/28804416/a66e83f3-b8a9-4858-8c09-e6157f2e6130)

**Reply by Banana71** [reply]
If the DAC is broken you have to unsolder it anyway. Then I would clean all the contacts nicely and connect 5 volts directly from a voltage source to the PINs VIN and GND and check whether the LED works or not. 
It's always a good idea to have a few DAC's at home. 😃.

**Reply by Banana71** [reply]
Is there 3.3 volts at contact (5) A3V3?

![DAC GY PCM5102 Schematic](https://github.com/probonopd/MiniDexed/assets/104296205/cca03753-3d70-4e40-a0c6-f41669779226)

**Comment by viragored** [comment]
Zero volts at A3V3, and 5V at VIN. 
Ali Express order is done. Let's hope the replacements work properly.
Thanks again for the help, it's much appreciated.

**Comment by viragored** [comment]
*********  Update  *********  
I've de-soldered the DAC. Putting USB power directly on VIN and GND does - - - nothing. I guess the board is properly dead.
Ali Express have just emailed to say that my order has shipped!

**Comment by viragored** [comment, marked as answer]
The problem was the board. Today I installed an AliExpress replacement and it's working perfectly.
And Ali set a new record for me - ordered on the 2nd, received on the 10th. Outstanding service by "WAVGAT Official Store"!
